db/seedData.js

```diff
@@ -0,0 +1,186 @@
+            DROP TABLE IF EXISTS billing_addresses;
+            DROP TABLE IF EXISTS puppies;
+            DROP TABLE IF EXISTS users;
+
+        `);
+        console.log("Finished dropping tables...")
+    } catch (error) {
+        console.error("Error dropping tables!");
+        throw error;
+    }
+}
+
+async function createTables() {
+    try {
+        console.log("Starting to build tables...");
+        await client.query(`
+            CREATE TABLE users (
+                id SERIAL PRIMARY KEY,
+                "firstName" VARCHAR(255) NOT NULL,
+                "lastName" VARCHAR(255) NOT NULL,
+                email VARCHAR(255) UNIQUE NOT NULL,
+                password VARCHAR(255) NOT NULL,
+                phone INTEGER,
+                "isActive" BOOLEAN DEFAULT true,
+                "isAdmin" BOOLEAN DEFAULT false
+            );
+        `);
+        await client.query(`
+            CREATE TABLE puppies (
+                id SERIAL PRIMARY KEY,
+                name VARCHAR(255) NOT NULL,
+                description TEXT NOT NULL,
+                age INTEGER,
+                breed VARCHAR(255) NOT NULL,
+                weight INTEGER,
+                size VARCHAR(255) NOT NULL,
+                pedigree BOOLEAN DEFAULT false,
+                "isVaccinated" BOOLEAN DEFAULT false,
+                "isNeutered" BOOLEAN DEFAULT false,
+                gender VARCHAR(255) NOT NULL,
+                "isAvailable" BOOLEAN DEFAULT true,
+                price INTEGER
+            );
+        `);
+        await client.query(`
+            CREATE TABLE billing_addresses (
+                id SERIAL PRIMARY KEY,
+                "userId" INTEGER REFERENCES users(id) NOT NULL,
+                street VARCHAR(255) NOT NULL,
+                city VARCHAR(255) NOT NULL,
+                state VARCHAR(255) NOT NULL,
+                zip INTEGER
+            );
+        `);
+        await client.query(`
+            CREATE TABLE shipping_addresses (
+                id SERIAL PRIMARY KEY,
+                "userId" INTEGER REFERENCES users(id) NOT NULL,
+                street VARCHAR(255) NOT NULL,
+                city VARCHAR(255) NOT NULL,
+                state VARCHAR(255) NOT NULL,
+                zip INTEGER
+            );
+        `);
+        await client.query(`
+            CREATE TABLE order_items (
+                id SERIAL PRIMARY KEY,
+                "userId" INTEGER REFERENCES users(id),
+                "puppyId" INTEGER REFERENCES puppies(id),
+                date TIMESTAMP,
+                status VARCHAR(255) NOT NULL,
+                UNIQUE ("userId", "puppyId")
+            );
+        `);
+        await client.query(`
+            CREATE TABLE cart_items (
+                id SERIAL PRIMARY KEY,
+                "userId" INTEGER REFERENCES users(id),
+                "puppyId" INTEGER REFERENCES puppies(id),
+                UNIQUE ("userId", "puppyId")
```

Comment:
Is this the way you had it in your fitnesstrackr and did it work? It's different from mine so I'm curious. Mine was:

CONSTRAINT uc_routine_activity UNIQUE ("routineId", "activityId")

which would be

CONSTRAINT uc_cart_item UNIQUE ("userId", "puppyId")

in this case.

Reply:
Yes, this was a bit that Michael helped me out with and it worked flawlessly.